pkg/sloop/processing/eventcount.go

```diff
@@ -50,26 +50,22 @@ func updateEventCountTable(
 
 	// Truncate long-lived events to available partitions
 	// This avoids filling in data that will go beyond the current time range
+	ok, minPartition, maxPartition := tables.GetMinAndMaxPartitionWithTxn(txn)
+	if err != nil || !ok {
+		return err
+	}
 
-	// Default truncate as computedLastTs -1 * PartitionDuration.
-	// Essentially only allowing events in 1 partition by default. This scenario will only be hit for first event on new sloop with no data.
-	truncateTs := computedLastTs.Add(-1 * untyped.GetPartitionDuration())
-
-	// If there is only one partition, use minPartitionStartTime to ensure we receive events
-	// in that partition.
-	// Otherwise add events to all partitions from MinPartitionEndTime to computedTs.
-	// This ensures no events are added to the very last partition which may get garbage collected soon.
-	if ok, minPartition, maxPartition := tables.EventCountTable().GetMinMaxPartitions(txn); ok {
-		if minPartitionStartTime, minPartitionEndTime, err := untyped.GetTimeRangeForPartition(minPartition); err == nil {
-			if minPartition == maxPartition {
-				truncateTs = minPartitionStartTime
-			} else {
-				truncateTs = minPartitionEndTime
-			}
-		}
+	_, minPartitionEndTime, err := untyped.GetTimeRangeForPartition(minPartition)
+	if err != nil && minPartition != "" {
```

Comment:
When there is no minpartition, the function itself just returns the error and does not continue. Empty partitionID is not handled as a special case as per say.

Reply:
Make sense. Testing here for minPartition to be empty is a useless check. I will remove it. Checking for error to be not nil is good enough. And since same logic applies to maxPartition, will do the same thing there too.